src/bindings/python/src/pyopenvino/graph/rt_map.cpp

```diff
@@ -27,19 +28,56 @@ using PyRTMap = ov::RTMap;
 
 PYBIND11_MAKE_OPAQUE(PyRTMap);
 
+// Create our custom iterator to return python object not OVAny itself.
+class PyRTMapIterator {
+public:
+    PyRTMapIterator(const PyRTMap& py_rt_map, py::object ref, bool is_value)
+        : py_rt_map(py_rt_map),
+          is_value(is_value),
+          ref(ref),
+          it(py_rt_map.cbegin()) {}
+
+    py::object next() {
+        if (it == py_rt_map.end()) {
+            throw py::stop_iteration();
+        }
+        const auto result = *it;
+        it++;
+        if (is_value) {
+            return Common::utils::from_ov_any_no_leaves(result.second);
+        } else {
+            std::pair<std::string, py::object> res = {result.first,
+                                                      Common::utils::from_ov_any_no_leaves(result.second)};
+            return py::cast(res);
+        }
+    }
+
+    const PyRTMap& py_rt_map;
+    bool is_value = false;
+    py::object ref;  // keep a reference
```

Comment:
```suggestion
    py::object ref;  // keep a reference
```
Isn't this a copy? Can it be switched to actual reference or is there some mechanism under `py::object` to do it when copy occurs?

Reply:
Discussed offline

Reply:
discussed offline